src/components/Api.js

```diff
@@ -0,0 +1,139 @@
+export default class Api {
+  constructor(data) {
+    this._authorization = "0863e235-ec04-4229-a6bf-890245ffa3f4";
+  }
+
+  /*uploadInitialCards(data) {
+    console.log(data);
+    data.forEach((data) => {
+      this.uploadCard(data);
+    });
+  }*/
+
+  uploadCard({ name, link }) {
+    return fetch("https://around-api.en.tripleten-services.com/v1/cards", {
+      method: "POST",
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: name,
+        link: link,
+      }),
+    }).then((res) =>
+      res.ok ? res.json() : Promise.reject(`Uh oh! Error: ${res.status}`)
+    );
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.